compiler/qsc_eval/src/backend.rs

```diff
@@ -13,32 +13,84 @@ use crate::val::Value;
 pub trait Backend {
     type ResultType;
 
-    fn ccx(&mut self, ctl0: usize, ctl1: usize, q: usize);
-    fn cx(&mut self, ctl: usize, q: usize);
-    fn cy(&mut self, ctl: usize, q: usize);
-    fn cz(&mut self, ctl: usize, q: usize);
-    fn h(&mut self, q: usize);
-    fn m(&mut self, q: usize) -> Self::ResultType;
-    fn mresetz(&mut self, q: usize) -> Self::ResultType;
-    fn reset(&mut self, q: usize);
-    fn rx(&mut self, theta: f64, q: usize);
-    fn rxx(&mut self, theta: f64, q0: usize, q1: usize);
-    fn ry(&mut self, theta: f64, q: usize);
-    fn ryy(&mut self, theta: f64, q0: usize, q1: usize);
-    fn rz(&mut self, theta: f64, q: usize);
-    fn rzz(&mut self, theta: f64, q0: usize, q1: usize);
-    fn sadj(&mut self, q: usize);
-    fn s(&mut self, q: usize);
-    fn swap(&mut self, q0: usize, q1: usize);
-    fn tadj(&mut self, q: usize);
-    fn t(&mut self, q: usize);
-    fn x(&mut self, q: usize);
-    fn y(&mut self, q: usize);
-    fn z(&mut self, q: usize);
-    fn qubit_allocate(&mut self) -> usize;
-    fn qubit_release(&mut self, q: usize);
-    fn capture_quantum_state(&mut self) -> (Vec<(BigUint, Complex<f64>)>, usize);
-    fn qubit_is_zero(&mut self, q: usize) -> bool;
+    fn ccx(&mut self, _ctl0: usize, _ctl1: usize, _q: usize) {
+        panic!("ccx is not supported");
```

Comment:
nitpick: these can be `unimplemented!` instead, since that matches the intent here.

Reply:
Updated.